### PR TITLE
[full-ci][tests-only] e2e test for spaces publiclink story

### PR DESCRIPTION
**Pull request by sushmita56**
## Description
This PR adds e2e test for public link spaces story.

## Related Issue
https://github.com/owncloud/web/issues/7407

## How Has This Been Tested?
- Locally
- On CI

### CODE REVIEW

File: tests/e2e/cucumber/features/integrations/spaces/publicLink.ocis.feature
Comment (SwikritiT) [inline tests/e2e/cucumber/features/integrations/spaces/publicLink.ocis.feature:20]:
```suggestion
        And  "Alice" creates the following resources
```

File: tests/e2e/cucumber/features/integrations/spaces/publicLink.ocis.feature
Comment (SwikritiT) [inline tests/e2e/cucumber/features/integrations/spaces/publicLink.ocis.feature:33]: And "Brian" logs out
And
```suggestion
        But "Brian" should not be able to edit the public link named "spaceLink"
        And "Brian" logs out
        And "Alice" logs out
```

File: tests/e2e/cucumber/steps/app-files/link.ts
Comment (kiranparajuli589) [inline tests/e2e/cucumber/steps/app-files/link.ts:43]: it's not good to add assertions inside the when step.

File: tests/e2e/support/objects/app-files/link/actions.ts
Comment (kiranparajuli589) [inline tests/e2e/support/objects/app-files/link/actions.ts:49]:
```suggestion
export type checkPublicLinkVisibilityArgs = {
```

File: tests/e2e/support/objects/app-files/link/actions.ts
Comment (kiranparajuli589) [inline tests/e2e/support/objects/app-files/link/actions.ts:208]:
```suggestion
export const getPublicLinkVisibility = async (
  args: checkpublicLinkVisibilityArgs
): Promise<void> => {
  const { page, linkUrl, linkName } = args
  await sidebar.open({ page: page })
  await sidebar.openPanel({ page: page, name: 'space-share' })
  await page.locator(util.format(publicLink, linkName)).textContent()
  // some calculations
  return true // or false
}
```
PO should only return the state of the Page we're traveling. Make the assertions inside the context (step definitions)

File: tests/e2e/support/objects/app-files/link/actions.ts
Comment (kiranparajuli589) [inline tests/e2e/support/objects/app-files/link/actions.ts:216]: see above comment

File: tests/e2e/cucumber/features/integrations/spaces/publicLink.ocis.feature
Comment (sushmita56) [reply tests/e2e/cucumber/features/integrations/spaces/publicLink.ocis.feature:32]: As I thought it would be difficult for reviewers to go through a bunch of changes across multiple files, I did not list all the changes in this PR. I will be writing a test for `Carol` too in the next PR.

File: tests/e2e/cucumber/steps/app-files/link.ts
Comment (kiranparajuli589) [inline tests/e2e/cucumber/steps/app-files/link.ts:127]:
```suggestion
    expect(isVisible).toBe(shouldOrShouldNot !== 'should not')
```
simplify

File: tests/e2e/cucumber/features/integrations/spaces/publicLink.ocis.feature
Comment (amrita-shrestha) [reply tests/e2e/cucumber/features/integrations/spaces/publicLink.ocis.feature]: @kiranparajuli589 and @sushmita56 I think we log out the user when all the action necessary for the scenario has been done by the user.

File: tests/e2e/cucumber/steps/app-files/link.ts
Comment (SwikritiT) [reply tests/e2e/cucumber/steps/app-files/link.ts:43]: we aren't using then steps unless it's absolutely necessary in e2e not our decision so assertions needs to be done on when step

https://github.com/owncloud/web/pull/7378#discussion_r936402653

Comment (amrita-shrestha) [reply tests/e2e/cucumber/steps/app-files/link.ts:43]: We don't use the **Then** step so assertions were done inside the **When** step  if need

File: tests/e2e/support/objects/app-files/link/actions.ts
Comment (kiranparajuli589) [inline tests/e2e/support/objects/app-files/link/actions.ts:57]: both types have the same keys inside. can we use one single type with a more generic name?

**Review by kiranparajuli589 — COMMENTED** [review]
just a small thing. rest lgtm :+1:

**Review by SagarGi — COMMENTED** [review]
works smoothly on my machine and code lgtm :+1:

**Comment by kulmann** [comment]
Please wait for a review from @fschade before merging

File: tests/e2e/support/objects/app-files/spaces/index.ts
Comment (fschade) [inline tests/e2e/support/objects/app-files/spaces/index.ts:119]: any reason why name ist static?

File: tests/e2e/support/objects/app-files/link/index.ts
Comment (fschade) [inline tests/e2e/support/objects/app-files/link/index.ts:85]: publicLinkVisible naming is misleading to me, can you tell what you want to achieve

File: tests/e2e/support/objects/app-files/link/index.ts
Comment (fschade) [inline tests/e2e/support/objects/app-files/link/index.ts:94]: same here, i would prefer a different naming like `buttonIsVisible` or similar... then it sounds more like a question which answers `yes` or `no`

File: tests/e2e/support/objects/app-files/link/actions.ts
Comment (fschade) [inline tests/e2e/support/objects/app-files/link/actions.ts:93]: seems like a leftover from the past... can you clean it up in your pr :) THX

File: tests/e2e/support/objects/app-files/link/actions.ts
Comment (fschade) [inline tests/e2e/support/objects/app-files/link/actions.ts:100]: is used only once here, can you please inline it

**Review by fschade — CHANGES_REQUESTED** [review]
thanks for the pr, please have a look at my comments.

File: tests/e2e/support/objects/app-files/spaces/index.ts
Comment (sushmita56) [reply tests/e2e/support/objects/app-files/spaces/index.ts:119]: Whenever we create a link from the UI level, the link gets created with the name `Link` by default no matter how many links we create for a single space. Link cannot be created with a specific name by using the web UI.  So, even if we pass the name dynamically from the feature file, the name won't be used actually on the UI level while automating the test. That's why I put a static name ` key: 'Link',` just to store the link information, and we do change this static name later on in the renaming step.

https://github.com/owncloud/web/blob/02a60335e006b8872e2eee7e19f427eb183eb51b/tests/e2e/support/objects/app-files/link/index.ts#L41-L52


Would you agree with this idea? Or should I make any changes? Any suggestions from your side?

File: tests/e2e/support/objects/app-files/link/actions.ts
Comment (sushmita56) [reply tests/e2e/support/objects/app-files/link/actions.ts:100]: The function could have been inlined, but I thought it would be better to create a separate function so that it can be reused in the future if needed.

File: tests/e2e/support/objects/app-files/link/index.ts
Comment (sushmita56) [reply tests/e2e/support/objects/app-files/link/index.ts:85]: It is to get the URL of the public link.

Comment (sushmita56) [reply tests/e2e/support/objects/app-files/link/index.ts]: It's now changed to `getPublicLinkUrl`

File: tests/e2e/support/objects/app-files/spaces/index.ts
Comment (fschade) [reply tests/e2e/support/objects/app-files/spaces/index.ts:119]: THX for clarification

File: tests/e2e/support/objects/app-files/link/actions.ts
Comment (fschade) [reply tests/e2e/support/objects/app-files/link/actions.ts:96]: Thanks, fine for me to have it as it is for later reuse :)

**Review by fschade — APPROVED** [review]
LGTM 🚀

**Pull request merged**
